File: PAT (Basic-Level)/C/1064.c
#include <stdio.h>

int main(int argc, char const *argv[])
{
    int N;
    scanf("%d", &N);
    int result[37] = {0};
    int temp;
    int friendnum = 0, count = 0;
    for (int i = 0; i < N; ++i)
    {
        friendnum = 0;
        scanf("%d", &temp);
        while(temp > 0){
            friendnum += (temp % 10);
            temp = temp / 10;
        }
        if(result[friendnum] == 0){
            count++;
            result[friendnum] = 1;
        }
    }
    printf("%d\n", count);
    for (int i = 0; i < 37; ++i)
    {
        if(result[i] == 1 && count != 1){
            printf("%d ", i);
            count--;
        }
        else if(result[i] == 1 && count == 1)
            printf("%d", i);
    }
    return 0;
}

File: PAT (Basic-Level)/Python/1064.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

N = int(input())
result = [0 for x in range(37)]
friendnum = count = 0
data = list(map(int, input().split()))

for i in data:
    friendnum = 0
    temp = i
    while temp > 0:
        friendnum += (temp%10)
        temp = int(temp / 10)
    if result[friendnum] == 0:
        count+=1
        result[friendnum] = 1

print(count)

for i in range(37):
    if result[i] == 1 and count != 1:
        print(i, end = ' ')
        count -= 1
    elif result[i] == 1 and count == 1:
        print(i)
